enhancements/authentication/pod-security-admission-enforcement.md

```diff
@@ -0,0 +1,419 @@
+In some other cases, the evaluation was impossible because PSA labels had been overwritten by users.
+Additional diagnostics are required to confirm the full set of potential causes.
+
+While the root causes need to be identified in some cases, the result of identifying a violating Namespace is understood.
+
+#### New SCC Annotation: `security.openshift.io/validated-scc-subject-type`
+
+The annotation `openshift.io/scc` currently indicates which SCC admitted a workload, but it does not distinguish **how** the SCC was granted — whether through a user or a Pod’s ServiceAccount.
+A new annotation will help determine if a ServiceAccount with the required SCCs was used, or if a user created the workload out of band.
+Because the PSA label syncer does not track user-based SCCs itself, it cannot fully assess labeling under those circumstances.
+
+To address this, the proposal introduces:
+
+```go
+// ValidatedSCCSubjectTypeAnnotation indicates the subject type that allowed the
+// SCC admission. This can be used by controllers to detect potential issues
+// between user-driven SCC usage and the ServiceAccount-driven SCC usage.
+ValidatedSCCSubjectTypeAnnotation = "security.openshift.io/validated-scc-subject-type"
+```
+
+This annotation will be set by the [`SecurityContextConstraint` admission](https://github.com/openshift/apiserver-library-go/blob/60118cff59e5d64b12e36e754de35b900e443b44/pkg/securitycontextconstraints/sccadmission/admission.go#L138) plugin.
+
+#### Set PSS Annotation: `security.openshift.io/MinimallySufficientPodSecurityStandard`
+
+The PSA label syncer must set the `security.openshift.io/MinimallySufficientPodSecurityStandard` annotation.
+Because users can modify `pod-security.kubernetes.io/warn` and `pod-security.kubernetes.io/audit`, these labels do not reliably indicate the minimal standard.
+The new annotation ensures a clear record of the minimal PSS that would be enforced if the `pod-security.kubernetes.io/enforce` label were set.
+
+#### Update the PodSecurityReadinessController
+
+By adding these annotations, the [`PodSecurityReadinessController`](https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/podsecurityreadinesscontroller/podsecurityreadinesscontroller.go) can more accurately identify potentially failing Namespaces and understand their root causes:
+
+- With the `security.openshift.io/MinimallySufficientPodSecurityStandard` annotation, it can evaluate Namespaces that lack the `pod-security.kubernetes.io/enforce` label but have user-overridden warn or audit labels.
+- With `ValidatedSCCSubjectType`, the controller can classify issues arising from user-based SCC workloads separately.
+  Many of the remaining clusters with violations appear to involve workloads admitted by user SCCs.
+
+
+### New API
+
+This API is used to support users to enforce PSA.
+As this is a transitory process, this API will loose its usefulness once PSA enforcement isn't optional anymore.
```

Comment:
Similar to previous comments I've made, I think it might be worth keeping this API around to allow cluster admins flexibility in how they configure the PodSecurity admission plugin. I think it is fine for us to have opinionated and encouraged defaults.

Have we thought through different use cases that admins may have and why they may or may not want to be able to configure the global enforcement mode?

Reply:
David's suggestion is to use LegacyConfig, which would infer that the field is introduced and deprecated immediately, and can then go through the deprecation cycle

Reply:
I can't imagine a use-case, where someone wouldn't want to have the highest possible security setting on their production cluster.
If I would imagine that I have a test-cluster, this could be annoying to deal with, if the legacyMode is gone. But it isn't like I could add a mechanism to auto-label namespaces with another PSS.